app/actions/websocket/system.ts

```diff
@@ -12,11 +14,11 @@ export async function handleLicenseChangedEvent(serverUrl: string, msg: WebSocke
 
     try {
         const license = msg.data.license;
-        const systems: IdValue[] = [{
-            id: SYSTEM_IDENTIFIERS.LICENSE,
-            value: JSON.stringify(license),
-        }];
+        const systems: IdValue[] = [{id: SYSTEM_IDENTIFIERS.LICENSE, value: JSON.stringify(license)}];
+
+        const prevLicense = await getConfig(operator.database);
         await operator.handleSystem({systems, prepareRecordsOnly: false});
+        checkDisplayNameSettings(serverUrl, license, prevLicense);
```

Comment:
here wrap in an if statement that checks if the LockTeammateNameDisplay property changed between the new license and the previous one.

Prefer the example above